source/numerics/smoothness_estimator.cc

```diff
@@ -0,0 +1,38 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 1998 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_handler.h>
```

Comment:
this file functionally only does `#include "smoothness_estimator.inst"` which only contains instantiations and no code. If you need all of these include files, then they should be `#include`d in the `.template.h` file.

Reply:
not all, just dof handler. That's a left-over from switch `templated DoFHandler` -> `hp::DoFHandler`.  The rest of headers are for instantiations for specific vectors types.